app/controllers/api/account.php

```diff
@@ -3469,39 +3500,44 @@
             ->dynamic($target, Response::MODEL_TARGET);
     });
 
-App::delete('/v1/account')
-    ->desc('Delete account')
+App::delete('/v1/account/targets/:targetId/push')
+    ->desc('Delete a push target')
     ->groups(['api', 'account'])
-    ->label('event', 'users.[userId].delete')
-    ->label('scope', 'accounts.write')
-    ->label('audits.event', 'user.delete')
-    ->label('audits.resource', 'user/{response.$id}')
-    ->label('usage.metric', 'users.{scope}.requests.delete')
-    ->label('sdk.auth', [APP_AUTH_TYPE_ADMIN])
+    ->label('scope', 'targets.write')
+    ->label('audits.event', 'target.delete')
+    ->label('audits.resource', 'target/response.$id')
+    ->label('event', 'users.[userId].targets.[targetId].delete')
+    ->label('sdk.auth', [APP_AUTH_TYPE_SESSION])
     ->label('sdk.namespace', 'account')
-    ->label('sdk.method', 'delete')
-    ->label('sdk.description', '/docs/references/account/delete.md')
+    ->label('sdk.method', 'deletePushTarget')
     ->label('sdk.response.code', Response::STATUS_CODE_NOCONTENT)
-    ->label('sdk.response.model', Response::MODEL_NONE)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_TARGET)
+    ->param('targetId', '', new UID(), 'Target ID.')
+    ->inject('queueForEvents')
+    ->inject('queueForDeletes')
     ->inject('user')
+    ->inject('request')
     ->inject('response')
     ->inject('dbForProject')
-    ->inject('queueForEvents')
-    ->inject('queueForDeletes')
-    ->action(function (Document $user, Response $response, Database $dbForProject, Event $queueForEvents, Delete $queueForDeletes) {
-        if ($user->isEmpty()) {
-            throw new Exception(Exception::USER_NOT_FOUND);
+    ->action(function (string $targetId, Event $queueForEvents, Delete $queueForDeletes, Document $user, Request $request, Response $response, Database $dbForProject) {
+        $target = Authorization::skip(fn() => $dbForProject->getDocument('targets', $targetId));
```

Comment:
How are we making sure a user is only deleting their own target?

Reply:
Addressed